File: app/src/main/java/de/mm/android/longitude/model/ProcessEntry.java
package de.mm.android.longitude.model;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

public class ProcessEntry {
    public static final String TAG = ProcessEntry.class.getSimpleName();
    public static final String TABLE = "t_process";
    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_LAT = "lat";
    public static final String COLUMN_LON = "lon";
    public static final String COLUMN_ALT = "alt";
    public static final String COLUMN_ACC = "acc";
    public static final String COLUMN_UPDATED_ON = "updated_on";
    public static final String COLUMN_ADDRESS = "address";

    private long id;
    private double lat;
    private double lon;
    private double alt;
    private double acc;
    private String updated_on;
    private String address;

    public ProcessEntry(double lat, double lon, double alt, double acc, String address) {
        this(-1, lat, lon, alt, acc, "", address);
    }

    public ProcessEntry(long id, double lat, double lon, double alt, double acc, String updated_on, String address) {
        this.id = id;
        this.lat = lat;
        this.lon = lon;
        this.alt = alt;
        this.acc = acc;
        this.updated_on = updated_on;
        this.address = address;
    }

    public long getId() {
        return id;
    }

    public double getLon() {
        return lon;
    }

    public double getLat() {
        return lat;
    }

    public double getAlt() {
        return alt;
    }

    public double getAccuracy() {
        return acc;
    }

    public String getUpdated_on() {
        return updated_on;
    }

    public String getAddress() {
        return address;
    }

    public double getAcc() {
        return acc;
    }

    public JSONObject toJSON() {
        try {
            JSONObject obj = new JSONObject();
            obj.put("id", id);
            obj.put("lat", lat);
            obj.put("lon", lon);
            obj.put("alt", alt);
            obj.put("acc", acc);
            obj.put("updated_on", updated_on);
            return obj;
        } catch (JSONException e) {
            Log.e(TAG, "JSONException", e);
            return null;
        }
    }

	/* Stuff */

    @Override
    public String toString() {
        return "ProcessEntry{" + "id=" + id + ", lat=" + lat + ", lon=" + lon + ", alt=" + alt +
                ", acc=" + acc + ", updated_on='" + updated_on + '\'' + ", address='" + address + '\'' + '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ProcessEntry that = (ProcessEntry) o;

        return id == that.id && Double.compare(that.lat, lat) == 0 && Double.compare(that.lon, lon) == 0 &&
                Double.compare(that.alt, alt) == 0 && Double.compare(that.acc, acc) == 0 &&
                updated_on.equals(that.updated_on) && !(address != null ?
                !address.equals(that.address) :
                that.address != null);

    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        result = (int) (id ^ (id >>> 32));
        temp = Double.doubleToLongBits(lat);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(lon);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(alt);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(acc);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        result = 31 * result + updated_on.hashCode();
        result = 31 * result + (address != null ? address.hashCode() : 0);
        return result;
    }
}

File: app/src/main/java/de/mm/android/longitude/model/ProcessEntry.kt
package de.mm.android.longitude.model

import android.util.Log

import org.json.JSONException
import org.json.JSONObject

class ProcessEntry(val id: Long, val lat: Double, val lon: Double, val alt: Double, val accuracy: Double, val updated_on: String, val address: String?) {

    constructor(lat: Double, lon: Double, alt: Double, acc: Double, address: String) : this(-1, lat, lon, alt, acc, "", address)

    fun toJSON(): JSONObject? {
        try {
            val obj = JSONObject()
            obj.put("id", id)
            obj.put("lat", lat)
            obj.put("lon", lon)
            obj.put("alt", alt)
            obj.put("acc", accuracy)
            obj.put("updated_on", updated_on)
            return obj
        } catch (e: JSONException) {
            Log.e(TAG, "JSONException", e)
            return null
        }

    }

    /* Stuff */

    override fun toString(): String {
        return "ProcessEntry{id=$id, lat=$lat, lon=$lon, alt=$alt, acc=$accuracy, updated_on='$updated_on\', address='$address\'}"
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as ProcessEntry?

        if (id != that.id) return false
        if (java.lang.Double.compare(that.lat, lat) != 0) return false
        if (java.lang.Double.compare(that.lon, lon) != 0) return false
        if (java.lang.Double.compare(that.alt, alt) != 0) return false
        if (java.lang.Double.compare(that.accuracy, accuracy) != 0) return false
        if (updated_on != that.updated_on) return false
        return !if (address != null) address != that.address else that.address != null

    }

    override fun hashCode(): Int {
        var result: Int
        var temp: Long
        result = (id xor id.ushr(32)).toInt()
        temp = java.lang.Double.doubleToLongBits(lat)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        temp = java.lang.Double.doubleToLongBits(lon)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        temp = java.lang.Double.doubleToLongBits(alt)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        temp = java.lang.Double.doubleToLongBits(accuracy)
        result = 31 * result + (temp xor temp.ushr(32)).toInt()
        result = 31 * result + updated_on.hashCode()
        result = 31 * result + if (address != null) address.hashCode() else 0
        return result
    }

    companion object {
        val TAG = ProcessEntry::class.java!!.getSimpleName()
        val TABLE = "t_process"
        val COLUMN_ID = "_id"
        val COLUMN_LAT = "lat"
        val COLUMN_LON = "lon"
        val COLUMN_ALT = "alt"
        val COLUMN_ACC = "acc"
        val COLUMN_UPDATED_ON = "updated_on"
        val COLUMN_ADDRESS = "address"
    }
}